tuf/repository_tool.py

```diff
@@ -2579,24 +2671,33 @@ def delegate_hashed_bins(self, list_of_targets, keys_of_hashed_bins,
         repr(prefix_count) + ' total hash prefixes.\n' +
         'Each bin ranges over ' + repr(bin_size) + ' hash prefixes.')
 
+    keyids, keydict = repo_lib.keys_to_keydict(keys_of_hashed_bins)
+
     # Generate a list of bin names, the range of prefixes to be delegated to
     # that bin, along with the corresponding full list of target prefixes
     # to be delegated to that bin
     ordered_roles = []
-    for idx in range(0, prefix_count, bin_size):
-      high = idx + bin_size - 1
-      name = repo_lib.create_bin_name(idx, high, prefix_length)
-      if bin_size == 1:
-        target_hash_prefixes = [name]
+    for idx in range(0, number_of_bins):
+      name = repo_lib.create_bin_name(idx, prefix_length, prefix)
+      if succinct:
+      # For succinct hased bin delegations, these metadata files will still
+      # be created, but they will not all be listed in the delegation
+        role = {"name":name,
+            "keyids": keyids,
+            "threshold": 1,
+            "terminating": False,
+            "target_paths": []}
+        ordered_roles.append(role)
```

Comment:
The role is already appended at the end of the loop, outside of the conditional. I think we're adding the role to ordered_roles twice?
```suggestion
```